### PR TITLE
Feature/glide 57

### PR DESCRIPTION
**Pull request by Horcon**
General tests encompassing the application should now for the most part cover the backend. Not testing error handling as I figure we won't be able to properly do that anyway.

### CODE REVIEW

File: rest/dbHandler.go [hunk 1]
```diff
@@ -127,7 +114,26 @@ func (dbHandler DbHandler) deleteTrackPage(w http.ResponseWriter, r *http.Reques
...
 	}
 
-	// Send response
-	w.WriteHeader(c)
+	w = prepareGeneralResponse(w, []byte(trackID), constant.TextPlain, nil, "deleteTrackPage")
+	return
+}
+
+// Prepares a general response to send back to the client, setting various common variables in the ResponseWriter
+func prepareGeneralResponse(unPrepW http.ResponseWriter, rawData []byte, contentType string, jsonError error, callerFunc string) (prepW http.ResponseWriter) {
```
Comment (MarkusAJacobsen) [inline rest/dbHandler.go]: Why does this method take in an error? Think you only call it with nil previously, and if an error is to be sent back to the user, why sent the whole error stack when you only use err.Error? e.i. the string version. If you manage to get rid of this param you will also drop down to the allowed param count

File: rest/dbHandler.go [hunk 2]
```diff
@@ -127,7 +114,26 @@ func (dbHandler DbHandler) deleteTrackPage(w http.ResponseWriter, r *http.Reques
...
+			Origin: fileNameDB,
+			Method: callerFunc + "-prepareGeneralResponse",
+			Err:    jsonError,
+		})
+
+		http.Error(prepW, jsonError.Error(), http.StatusBadRequest)
+		return
+	}
+
+	prepW.Header().Set("Content-Type", contentType)
```
Comment (MarkusAJacobsen) [inline rest/dbHandler.go]: Just a note, I think we have a const of "Content-Type"

File: rest/dbHandler_test.go [hunk 3]
```diff
@@ -54,5 +42,94 @@ func TestDbHandler(t *testing.T) {
...
+	}
+	// insertTrack DONE
 
+	// Set up getTracks
+	req = httptest.NewRequest("GET", "/getTracks", nil)
+	req.Header.Set("token", token)
+	req.Header.Set("queryType", "Private")
+	req.Header.Set("orderDirection", "Desc")
+
+	// Run getTracks
```
Comment (MarkusAJacobsen) [inline rest/dbHandler_test.go:79]: If you are to do multiple steps in a test which relies on each other, consider using t.Run("", func()) instead, as you can skip the rest if something fails, as well as separating the logic.

**Review by MarkusAJacobsen — CHANGES_REQUESTED** [review]

File: rest/dbHandler.go (hunk 1 above)
Comment (Horcon) [reply rest/dbHandler.go]: The reason I do this is to cut down on the repeated code basically when it comes to error handling. If I remove that there we'll need that error handling code to basically be there in each of the 4 functions instead. It's used when we parse json in getTracks and insertTrack, as the parsing can give an error and is then handled by the response code. I've moved the error code back up, we may make it a new task to improve the look of the current error logging, the current way of logging messages feels rather clunky.

File: rest/dbHandler.go (hunk 2 above)
Comment (Horcon) [reply rest/dbHandler.go]: Oh yeah we do.

File: rest/dbHandler_test.go (hunk 3 above)
Comment (Horcon) [reply rest/dbHandler_test.go:79]: While noted, in the event that getTracks or getTrack fail, that would make deleteTrack not run, not cleaning up the database from the insertTrack. In addition, each step provides variables that the next test depends on, which is also the point of the test, confirming that everything works as expected. Tried creating 2 t.Run functions, making a variable in one and trying to make use of it in another, and I get errors on trying to use it. Tried quickly to create it outside of the functions, setting in 1, then trying to use it in 2, and the error message came up when I tried to check whether it didn't hold what was set in 1, meaning setting the variables like that won't work. So, the way it's set up is actually necessary.

**Review by MarkusAJacobsen — APPROVED** [review]

**Pull request merged**